site/app/libraries/Core.php

```diff
@@ -197,7 +198,12 @@ public function getQueries() {
     public function loadUser($user_id) {
         // attempt to load rcs as both student and user
         $this->user_id = $user_id;
-        $this->user = $this->database_queries->getUserById($user_id);
+        if($this->getConfig()->getCourse() === "" || $this->getConfig()->getCourse() === ""){
```

Comment:
same check twice?

However, I will make a comment here that applies throughout which is that I don't really like checking for `getCourse()` to be empty to signify that we haven't loaded a course. I'd rather we add a boolean `$loaded_course` to the Config model that defaults to false and we set to true if we have an `.ini` file to load (basically we get into the loadCourseIni function) and then we can use the magic `__call` function to just have `hasLoadedCourse()` (add this as a @method to the top of the class) instead of these `getCourse() === ""` checks. (and any time we check for if `getCourse()` or `getSemester()` are blank, we can just use the one `hasLoadedCourse()` function as well since it would nicely cover both cases).